code/game/machinery/autolathe.dm

```diff
@@ -136,93 +358,131 @@
+	for(var/MAT in being_built.materials)
+		var/datum/material/used_material = MAT
+		var/amount_needed = being_built.materials[MAT] * coeff * multiplier
+		if(istext(used_material)) //This means its a category
+			used_material = build_queue[requested_design_id]["build_mat"]
+			if(!used_material)
+				build_queue -= requested_design_id
+				item_queue -= requested_design_id
+				addtimer(CALLBACK(src, .proc/restart_process), 50)
+				return //Didn't pick any material, so you can't build shit either.
+			custom_materials[used_material] += amount_needed
 
-		if(href_list["search"])
-			matching_designs.Cut()
+		materials_used[used_material] = amount_needed
 
-			for(var/v in stored_research.researched_designs)
-				var/datum/design/D = SSresearch.techweb_design_by_id(v)
-				if(findtext(D.name,href_list["to_search"]))
-					matching_designs.Add(D)
-			updateUsrDialog()
+	if(materials.has_materials(materials_used))
+		busy = TRUE
+		use_power(power)
+		icon_state = "autolathe_n"
+		var/time = is_stack ? 32 : (32 * coeff * multiplier) ** 0.8
+		//===Repeating mode===
+		//Remove from queue
+		if(from_build_queue)
+			build_queue[requested_design_id]["amount"] -= multiplier
+			if(build_queue[requested_design_id]["amount"] <= 0)
+				build_queue -= requested_design_id
+		else
+			var/list/queue_data = item_queue[requested_design_id]
+			item_queue[requested_design_id]["amount"] -= multiplier
+			var/removed = FALSE
+			if(item_queue[requested_design_id]["amount"] <= 0)
+				item_queue -= requested_design_id
+				removed = TRUE
+			//Requeue if necessary
+			if(queue_repeating || queue_data["repeating"])
+				stored_item_amount ++
+				if(removed)
+					add_to_queue(item_queue, requested_design_id, stored_item_amount, queue_data["repeating"])
+					stored_item_amount = 0
+		//Create item and restart
+		being_build_finish_time = world.time + time
+		total_build_time = time
+		addtimer(CALLBACK(src, .proc/make_item, power, materials_used, custom_materials, multiplier, coeff, is_stack), time)
+		addtimer(CALLBACK(src, .proc/restart_process), time + 5)
 	else
-		to_chat(usr, "<span class=\"alert\">The autolathe is busy. Please wait for completion of previous operation.</span>")
-
-	updateUsrDialog()
-
-	return
+		say("Insufficient materials, operation will proceed when sufficient materials are available.")
+		operating = FALSE
+		wants_operate = TRUE
+
+/obj/machinery/autolathe/proc/restart_process()
+	operating = FALSE
+	wants_operate = FALSE
+	if(disabled)
+		return
+	begin_process()
 
 /obj/machinery/autolathe/proc/make_item(power, var/list/materials_used, var/list/picked_materials, multiplier, coeff, is_stack)
+	if(QDELETED(src))
+		return
+	//Stops the queue
+	if(disabled)
+		operating = FALSE
+		return
 	var/datum/component/material_container/materials = GetComponent(/datum/component/material_container)
-	var/atom/A = drop_location()
+	var/turf/A = get_release_direction()
```

Comment:
Should probably be get_release_turf since that's really what it returns

Reply:
Done